trial-config-service/src/test/java/uk/ac/ox/ndph/mts/trialconfigservice/controller/TrialConfigControllerTest.java

```diff
@@ -0,0 +1,74 @@
+package uk.ac.ox.ndph.mts.trialconfigservice.controller;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ox.ndph.mts.trialconfigservice.config.WebConfig;
+import uk.ac.ox.ndph.mts.trialconfigservice.exception.InvalidConfigException;
+import uk.ac.ox.ndph.mts.trialconfigservice.model.Trial;
+import uk.ac.ox.ndph.mts.trialconfigservice.model.TrialRepository;
+import uk.ac.ox.ndph.mts.trialconfigservice.model.TrialSite;
+import uk.ac.ox.ndph.mts.trialconfigservice.service.TrialConfigService;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class TrialConfigControllerTest {
+
+    private static final String TEST_CONFIG_ENDPOINT = "anyurlasitsmocked";
+
+    private TrialConfigController trialConfigController;
+
+    @Mock
+    private TrialRepository trialRepository;
+
+    @Mock
+    private TrialConfigService trialConfigService;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    WebConfig webConfig;
+
+    @BeforeEach
+    void setUp() {
+        trialConfigController = new TrialConfigController(trialConfigService, trialRepository, webConfig);
+    }
+
+    @Test
+    void createTrialFromJsonFile() {
```

Comment:
I would recommend the following naming convention for unit tests:
`<function name>_GiveACondition_ThenExpectedResult`
This may help others to read the test
Alternatively, lets agree on a naming convention for unit test applied by all developers